sdk/src/realtime/composable.ts

```diff
@@ -239,7 +248,10 @@ export function realtime(config: WebSocketConfig = {}) {
 				collection: Collection,
 				options = {} as Options,
 			) {
-				if (!socket || socket.readyState !== WebSocketState.OPEN) await this.connect();
+				if (!socket || !isConnected) {
+					throw new Error('Cannot subscribe without an open connection. Make sure you are calling "await client.connect()".');
```

Comment:
Can the `subscribe` call automatically call `await connect()` on the user's behalf? Is a bit better DX than having to manually call connect first.

Reply:
I do agree that will be better DX however with the current approach of calling connect the errors seem to get hidden when something goes wrong during connecting like bad credentials, timeouts, being unable to connect etc. This in turn leads to a lot of question at the moment about why "subscriptions" are not firing which always seems to be connection/authentication errors without errors to the user.

Reply:
I can do another dive into the issue after my other tasks to see if i can get it to properly throw these errors instead of removing the connect feature 🤔

Reply:
Would the subscribe call itself not reject in that case?

So you'd do something like

```ts
try {
  await subscribe('x', callback);
} catch (err) {
  // connection error
}
```

Reply:
@br41nslug Maybe the return of subscribe is the underlying connection, which can be an event emitter?

```ts
const subscription = await subscribe('x', callback);

subscription.on('connectionError', errorCallback);
```

?

Reply:
@rijkvanzanten The client itself is already an event emitter where you can listen for low level connection events.

```js
client.onWebsocket('open', () => {})
client.onWebsocket('error', () => {})
client.onWebsocket('close', () => {})
client.onWebsocket('message', () => {})
```